### DISCUSSION TITLE
g2@4.2.4 can't show legend

### DISCUSSION THREAD

**Discussion by chiga0** [antvis/G2#4993] [Q&A] [answered]
- [x] I have searched the [issues](https://github.com/antvis/g2/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/flamboyant-poitras-n1qykn?file=/index.tsx)

### Steps to reproduce
as the code shows, I set the legend visible, but it doesn't show.
I had set the line size, line color and line shape attribute.
and I had tried followings:
- `chart.legend(true)`
- `chart.legend({ position: 'bottom'})`
- add bottom padding

| Environment | Info |
|---|---|
| g2 | 4.2.4 |
| System | - |
| Browser | - |

**Comment by chiga0** [comment]
尝试g2降级到3.x，上述写法 图例是完美展示的。
目前可以确定确实是升级4.x带来的问题，所以不清楚是 4.x底层默认不支持 双轴图的图例展示， 还是 4.x升级时这部分逻辑漏掉了。
如果设计不支持，建议升级文档能够说明下，毕竟也算是 break change 了。

补充一个3.x的demo：https://codesandbox.io/s/young-mountain-x1u73c

**Reply by chiga0** [reply]
能否在g2中内置这部分逻辑？因为我们这边是直接使用g2，而不是g2plot~。
如果升级g2plot的话工作量会很大，这个需要评估下。

**Comment by hustcc** [comment, marked as answer]
在 G2 V3 中，内置了处理双轴图的逻辑，自动生成图例，但是实际上有挺多问题，所以在 V4 中，抽你出来，这部分逻辑封装到 G2Plot 中，可以看这个 [DEMO](http://g2plot.antv.antgroup.com/examples/dual-axes/column-line/#column-line)

建议直接使用 [G2Plot](https://g2plot.antv.antgroup.com/) 的组件，默认就处理了双轴图的图例问题，处理的比 v3 肯定更好。

![image](https://github.com/antvis/G2/assets/7856674/88ef86c2-6e48-451b-8f58-6c15841c6826)
